tests/qemu/linux/test_qemu_platform_detail.cpp

```diff
@@ -0,0 +1,202 @@
+            .WillOnce(Return(false))
+            .WillOnce(Return(true));
+
+        EXPECT_CALL(*mock_file_ops, open(_, _)).WillRepeatedly(Return(true));
+        EXPECT_CALL(*mock_file_ops, write(_, _)).WillRepeatedly(Return(1));
+    };
+
+    mpt::TempDir data_dir;
+    const QString multipass_bridge_name{"mpqemubr0"};
+    const std::string hw_addr{"52:54:00:6f:29:7e"};
+    const std::string subnet{"192.168.64"};
+    const std::string name{"foo"};
+
+    std::unique_ptr<mpt::MockDNSMasqServer> mock_dnsmasq_server;
+    std::unique_ptr<mpt::MockFirewallConfig> mock_firewall_config;
+
+    mpt::MockUtils::GuardedMock utils_attr{mpt::MockUtils::inject<NiceMock>()};
+    mpt::MockUtils* mock_utils = utils_attr.first;
+
+    mpt::MockBackend::GuardedMock backend_attr{mpt::MockBackend::inject<NiceMock>()};
+    mpt::MockBackend* mock_backend = backend_attr.first;
+
+    mpt::MockDNSMasqServerFactory::GuardedMock dnsmasq_server_factory_attr{
+        mpt::MockDNSMasqServerFactory::inject<NiceMock>()};
+    mpt::MockDNSMasqServerFactory* mock_dnsmasq_server_factory = dnsmasq_server_factory_attr.first;
+
+    mpt::MockFirewallConfigFactory::GuardedMock firewall_config_factory_attr{
+        mpt::MockFirewallConfigFactory::inject<NiceMock>()};
+    mpt::MockFirewallConfigFactory* mock_firewall_config_factory = firewall_config_factory_attr.first;
+
+    mpt::MockFileOps::GuardedMock file_ops_attr{mpt::MockFileOps::inject<NiceMock>()};
+    mpt::MockFileOps* mock_file_ops = file_ops_attr.first;
+
+    mpt::MockLogger::Scope logger_scope = mpt::MockLogger::inject();
+};
+} // namespace
+
+TEST_F(QemuPlatformDetail, ctor_sets_up_expected_virtual_switch)
+{
+    const QString qstring_subnet{QString::fromStdString(subnet)};
+
+    EXPECT_CALL(*mock_utils, run_cmd_for_status(QString("ip"),
+                                                ElementsAre(QString("link"), QString("add"), multipass_bridge_name,
+                                                            QString("address"), _, QString("type"), QString("bridge")),
+                                                _))
+        .WillOnce(Return(true));
+    EXPECT_CALL(*mock_utils, run_cmd_for_status(QString("ip"),
+                                                ElementsAre(QString("address"), QString("add"),
+                                                            QString("%1.1/24").arg(qstring_subnet), QString("dev"),
+                                                            multipass_bridge_name, "broadcast",
+                                                            QString("%1.255").arg(qstring_subnet)),
+                                                _))
+        .WillOnce(Return(true));
+    EXPECT_CALL(*mock_utils, run_cmd_for_status(
+                                 QString("ip"),
+                                 ElementsAre(QString("link"), QString("set"), multipass_bridge_name, QString("up")), _))
+        .WillOnce(Return(true));
```

Comment:
Could we get rid of `QString()` all over the place?

Reply:
Not really. The matchers for EXPECT_CALL functions on a mock don't really follow what the arguments are for a mock. For example, here I already have something like:
```
MOCK_CONST_METHOD3(run_cmd_for_status, bool(const QString&, const QStringList&, const int));
...
EXPECT_CALL(*mock_utils, run_cmd_for_status("ip", ElementsAre("fe", "fi, "fo", "fum"), _));
```

GoogleMock fails to understand that these are supposed to be QString's even though the MOCK_CONST_METHOD specifies it as such. This was the only way I found it to work.